src/reporter.ts

```diff
@@ -136,6 +136,10 @@ export default class SauceReporter extends SummaryFormatter {
     this.passed = testRunFinished.success;
     this.reportToFile(this.testRun)
 
+    if (!this.shouldUpload) {
```

Comment:
And who sets this setting? 🤔

Reply:
This args is parsed from cucumber `formatOptions` and set in constructor https://github.com/saucelabs/sauce-cucumber-reporter/blob/main/src/reporter.ts#L55

Reply:
Aye, but who ought to set this value? The user?

Reply:
hmm.. should be the user. I set it to false when it's on sauce VM.

Reply:
The user should set it to false when running on sauce? I don't think it's a good idea to give control over this behavior to the user. It's fine if the user wants to disable upload to sauce when running locally though.

Our VMs should have an environment variable that indicates that it's a Sauce VM.

Our runners do the same check to prevent unnecessary report creations (`process.env.SAUCE_VM`). I think the reporter here should do the same.

Reply:
> The user should set it to false when running on sauce
I mean the sauce runner is also a user to use the reporter. So I just set this option to false in the runner. https://github.com/saucelabs/sauce-playwright-runner/pull/171/files#diff-26afe8aaf5da7f0565526683694bf4715e308fe71ec1daf42fe150e4dd61065bR116

I also set the option to false when detecting the `SAUCE_VM` env var in the coming commit.

Reply:
Ahh perfect 👍